src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -329,14 +329,27 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
   // same TaskID to request a worker
   auto resource_spec_msg = scheduling_key_entry.resource_spec.GetMutableMessage();
   resource_spec_msg.set_task_id(TaskID::FromRandom(job_id_).Binary());
-  const TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
+  TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
   rpc::Address best_node_address;
   const bool is_spillback = (raylet_address != nullptr);
+  float scheduler_spread_threshold = RayConfig::instance().scheduler_spread_threshold();
   if (raylet_address == nullptr) {
+    bool is_selected_based_on_locality;
     // If no raylet address is given, find the best worker for our next lease request.
-    best_node_address = lease_policy_->GetBestNodeForTask(resource_spec);
+    std::tie(best_node_address, is_selected_based_on_locality) =
```

Comment:
Shouldn't we only run this logic if the scheduling strategy is DEFAULT?

Reply:
Currently that check is inside lease_policy: it checks the scheduling strategy and decide if it needs to run locality logic.